### PR TITLE
Scripting

### PR DESCRIPTION
**Pull request by fanick1**
- should look good

- integrates AngelScript library into the game for scripting support 

- per-map scripts (<map_name>.as in /levels folder alongside the map)
- global script - levels/global.as
    - enables raising the water when players wander around the level and don't kill each other (interval is proportional to number of players currently alive)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,7 +32,7 @@ if(MINGW)
 endif(MINGW)
 
 # platform-independent debug macro
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DD6_DEBUG")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DD6_DEBUG -ggdb3 -g")
```
Comment (odanek) [inline CMakeLists.txt]: K cemu je to dobre?

**Comment by odanek** [comment]
Neuveritelne mnozstvi kodu :D Co by se mi libilo by bylo minimalne udelat cele skriptovani volitelne pripadne jeste lepe umoznit ruzne skriptovaci enginy. Takze clovek by si v CMake zvolil jake skriptovani chce (AngelScript, LUA, JS, co ja vim...) a to by se mu zbuildovalo. K tomu bych tam udelal nejake obecne interfacy a pro ne pak implementace pro jednotlive skriptovaci enginy. Veci specificke pro angelscript bych pak hodil do adresare /script/as. Ted je to tam hodne provazane a kdyby si clovek chtel zkusit jiny engine, tak je celkem nahrany.

File: CMakeLists.txt (hunk 1 above)
Comment (fanick1) [inline CMakeLists.txt]: tohle se nemelo dostat do commitu :) odstranim

**Comment by fanick1** [comment]
Promyslim to :)

**Pull request closed without merging**